azure/infrastructure.tf

```diff
@@ -60,6 +69,7 @@ resource "azurerm_linux_virtual_machine" "instances" {
   location              = var.location
   resource_group_name   = local.resource_group_name
   network_interface_ids = [azurerm_network_interface.nic[each.key].id]
+  availability_set_id   = contains(each.value["tags"], "node") ? azurerm_availability_set.avset.id : null
```

Comment:
This is just an example of how this could be done, perhaps you'd prefer an additional tag?